assets/src/dashboard/app/views/editorSettings/googleAnalytics/index.js

```diff
@@ -55,47 +57,67 @@ export const TEXT = {
 function GoogleAnalyticsSettings({ googleAnalyticsId, handleUpdate }) {
   const [analyticsId, setAnalyticsId] = useState(googleAnalyticsId);
   const [inputError, setInputError] = useState('');
+  const canSave = analyticsId !== googleAnalyticsId && !inputError;
+  const disableSaveButton = !canSave;
 
   useEffect(() => {
     setAnalyticsId(googleAnalyticsId);
   }, [googleAnalyticsId]);
 
-  const handleCancelUpdateId = useCallback(() => {
-    setAnalyticsId(googleAnalyticsId);
-  }, [googleAnalyticsId, setAnalyticsId]);
+  const handleUpdateId = useCallback((event) => {
+    const { value } = event.target;
+    setAnalyticsId(value);
+
+    if (value.length === 0 || validateGoogleAnalyticsIdFormat(value)) {
+      setInputError('');
+
+      return;
+    }
 
-  const handleUpdateId = useCallback(
-    (value) => {
-      if (value.length === 0 || validateGoogleAnalyticsIdFormat(value)) {
-        setInputError('');
-        return handleUpdate(value);
+    setInputError(TEXT.INPUT_ERROR);
+  }, []);
+
+  const handleOnSave = useCallback(() => {
+    if (canSave) {
+      handleUpdate(analyticsId);
+    }
+  }, [canSave, analyticsId, handleUpdate]);
+
+  const handleOnKeyDown = useCallback(
+    (e) => {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        handleOnSave();
       }
-      return setInputError(TEXT.INPUT_ERROR);
     },
-    [handleUpdate, setInputError]
+    [handleOnSave]
   );
+
   return (
     <SettingForm onSubmit={(e) => e.preventDefault()}>
       <SettingHeading htmlFor="gaTrackingID">
         {TEXT.SECTION_HEADING}
       </SettingHeading>
       <FormContainer>
-        <InlineInputForm
+        <TextInput
           label={TEXT.ARIA_LABEL}
           id="gaTrackingId"
           value={analyticsId}
-          onEditCancel={handleCancelUpdateId}
-          onEditComplete={handleUpdateId}
+          onChange={handleUpdateId}
+          onKeyDown={handleOnKeyDown}
           placeholder={TEXT.PLACEHOLDER}
           error={inputError}
```

Comment:
you can scrap this error prop, it was to render within the `InlineInputForm` element

Reply:
`error` is actually used by `TextInput` and was being passed down to it through `InlineInputForm`

```javascript
export const TextInput = styled.input`
  ${TypographyPresets.Small};
  width: 100%;
  margin: 0;
  padding: 1px 8px;
  border-radius: 6px;
  border: ${({ theme, error }) =>
    error ? theme.borders.danger : theme.borders.gray100};
  &:active {
    border: ${({ theme, error }) =>
      error ? theme.borders.danger : theme.borders.action};
  }
`;
```

Reply:
oh! my bad!